pkg/gitproviders/provider.go

```diff
@@ -436,6 +438,47 @@ func (p defaultGitProvider) CreatePullRequestToOrgRepo(orgRepRef gitprovider.Org
 	return pr, nil
 }
 
+func (p defaultGitProvider) GetCommitsFromUserRepo(userRepRef gitprovider.UserRepositoryRef, targetBranch string) ([]gitprovider.Commit, error) {
+	ctx := context.Background()
+
+	ur, err := p.provider.UserRepositories().Get(ctx, userRepRef)
+	if err != nil {
+		return nil, fmt.Errorf("error getting info for repo [%s] err [%s]", userRepRef.String(), err)
+	}
+
+	// currently locking the commit list at 10. May discuss pagination options later.
+	commits, err := ur.Commits().ListPage(ctx, targetBranch, 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("error getting commits for repo[%s] err [%s]", userRepRef.String(), err)
+	}
+
+	if len(commits) == 0 {
+		return nil, fmt.Errorf("targetBranch[%s] has no commits", targetBranch)
+	}
+
+	return commits, nil
+}
+
+func (p defaultGitProvider) GetCommitsFromOrgRepo(orgRepRef gitprovider.OrgRepositoryRef, targetBranch string) ([]gitprovider.Commit, error) {
+	ctx := context.Background()
+
+	ur, err := p.provider.OrgRepositories().Get(ctx, orgRepRef)
+	if err != nil {
+		return nil, fmt.Errorf("error getting info for repo [%s] err [%s]", orgRepRef.String(), err)
+	}
+
+	// currently locking the commit list at 10. May discuss pagination options later.
+	commits, err := ur.Commits().ListPage(ctx, targetBranch, 10, 0)
```

Comment:
nit - the function will be more generic if the caller can specify the number of commits to return. Of course, that will mean guarding against a really large number.

Reply:
The decision was made to lock it at 10. Figured it would be safer to not allow it until we decided what to do next.